### ISSUE TITLE
Probable bug with skimage.transform.rotate using "center" kwarg

### ISSUE THREAD

**Issue by leducvin** [scikit-image/scikit-image#2671]
## Description
There seems to be a bug with the center keyword argument in the skimage.transform.rotate function. Well, either a bug or an error in the documentation.

For this keyword, the documentation says: 

> The rotation center. If `center=None`, the image is rotated around its center, i.e. `center=(rows / 2 - 0.5, cols / 2 - 0.5)`.

This seems to imply that, if `center` is not `None`, it should be of value `(row, col)` to choose the center of rotation.

I'm pretty sure the documentation should say `center=(cols / 2 - 0.5, rows / 2 - 0.5)`. Or the code needs to be changed.

The current tests for the `rotate` function in `skimage.transform.tests.test_warps.py` are not sensitive to this bug, since they are all using centers on the diagonal of an array (i.e. `row` = `col`).

## Way to reproduce
Just run this, which, minus the figures, could be used as a test:

```
import matplotlib.pyplot as plt
import numpy
import skimage.transform
img = numpy.array(
    [
        [0, 0, 0, 0, 0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0, 0, 0, 0, 0],
        [0, 0, 0, 1, 1, 1, 0, 0, 0],
        [0, 0, 2, 1, 0, 1, 0, 0, 0],
        [0, 0, 0, 1, 1, 1, 0, 0, 0],
        [0, 0, 0, 0, 0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0, 0, 0, 0, 0],
        [0, 0, 0, 0, 0, 0, 0, 0, 0],
    ]
)

fig = plt.figure()
plt.imshow(img)
plt.title('Original image')

img_rot_center = skimage.transform.rotate(
    img, 90, order=0, preserve_range=True)

fig = plt.figure()
plt.imshow(img_rot_center)
plt.title('Image rotated 90 degrees CCW around center')

# This works fine, value has not changed at the center of the image
assert img[5, 5] == img_rot_center[5, 5]

# Try to rotate around row=4, col=2, which is the value set at 2 in the array.
img_rot_off_center = skimage.transform.rotate(
    img, 90, order=0, center=(4, 2), preserve_range=True)

fig = plt.figure()
plt.imshow(img_rot_off_center)
plt.title('Image rotated 90 degrees CCW around row=4, col=2')
plt.show()

# The value of pixel around which rotation is done should not change.
# Currently this does not pass, unless rotate is called with
# center=(2,4)
assert img[4, 2] == img_rot_off_center[4, 2]
```

I'm on python 2.7 / Ubuntu 16.04 / skimage 0.13.0.

**Comment by soupault** [comment]
/xref https://github.com/scikit-image/scikit-image/issues/1732

**Comment by jpangburn** [comment]
This is kind of a nasty bug, because depending on your rotation it may only be off by a little ways so some images will look right (where row and column values were close) while other images will look wrong (where row and column values differed more).  So it could eat up a lot of time for someone to determine that while everything else in the library is (r,c), this center parameter is (c,r).

I would suggest that the documentation is fixed as the OP requested, and put a bold comment in the notes because this is a deviation from what the user expects.

**Comment by jni** [comment]
@jpangburn thanks for the ping. Would you be willing to submit a pull request fixing this? In the medium term, though, we are aiming to deprecate the coordinate transposition throughout `transform`, which should eliminate this kind of bug once and for all.

**Comment by jpangburn** [comment]
@jni Yes, I'd be happy to.  I had looked in the source code and seen this was caused by the transform operating with (c,r), and saw that same center documented correctly for the swirl method.  After noticing this, I thought maybe it's just cause I'm new to using this tool, and maybe the (c,r) usage was more common and only fooled me.  So I didn't submit a pull request.

**Comment by jpangburn** [comment]
Pull request completed [3341](https://github.com/scikit-image/scikit-image/pull/3341)
